salt/modules/boto_asg.py

```diff
@@ -141,7 +141,7 @@ def get_config(name, region=None, key=None, keyid=None, profile=None):
     retries = 30
     while True:
         try:
-            asg = [g for g in conn.get_all_groups(names=[name]) if g.name]
+            asg = [g for g in conn.get_all_groups(names=[name]) if all([g.name, g.vpc_zone_identifier])]
```

Comment:
it's a bit strange that the name check would fail to identify problem rows though, right??

Reply:
Yeah I thought so to, but alas, `vpc_zone_identifier` is apparently `None`. I can only assume that at least one of these dodgy groups has a truthy name, but a `None` for `vpc_zone_identifier`, and that this hasn't been the case in the past.